samples/net/gsm_modem/src/main.c

```diff
@@ -20,7 +20,7 @@ LOG_MODULE_REGISTER(sample_gsm_ppp, LOG_LEVEL_DBG);
 #define GSM_MODEM_NODE DT_COMPAT_GET_ANY_STATUS_OKAY(zephyr_gsm_ppp)
 #define UART_NODE DT_BUS(GSM_MODEM_NODE)
 
-static const struct device *const gsm_dev = DEVICE_DT_GET(GSM_MODEM_NODE);
+static const struct device *const gsm_dev = DEVICE_DT_GET_OR_NULL(GSM_MODEM_NODE);
```

Comment:
What about `DEVICE_DT_GET_ONE` so that it fails to compile if there's none in the dts?

Reply:
I just wanted to allow compiling the sample even if there is no node in the dts and return an error at the runtime. Do you think is it a bad idea?

Reply:
This change seems to be in conflict with another PR (https://github.com/zephyrproject-rtos/zephyr/pull/49638), which will cause a compilation error if the node is not found. We need to decide what's more convenient to have here (clear compile-time error or runtime error). I'd personally lean towards compile-time error in case of improper DT configuration, but no strong feelings here.

Another thing also pointed in that PR is that the sample documentation got a bit outdated now, the default suggested build command with `reel_board` no longer works. @ycsin Do you have a proven, working configuration that we could use in the docs perhaps as reference?

Reply:
Yup, that's true. I'm using ```west build -p always -d test_gsm -b frdm_k64f samples/net/gsm_modem -- -DCONFIG_MODEM_GSM_APN=\"internet\" -DDTC_OVERLAY_FILE="boards/frdm_uart2_dts.overlay"``` as it's pointed in the sample.yaml file.

Reply:
Now I think that a compile-time error would be better. Maybe I should just remove the change related to the `DEVICE_DT_GET` macro and keep the pointers checks only since we use `DEVICE_DT_GET` macro. What do you think?

Reply:
Would that produce a compile time error though? What about @ycsin suggestion, it should end up in an error compile-time according to the documentation (not during the linking stage).

Reply:
I remember I read somewhere that the DT_DEVICE_GET is for driver, for application _GET_ONE _GET_ANY should be used instead. Probably @mbolivar has an insight for this?